lib/do_image/do_update.js

```diff
@@ -0,0 +1,146 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2020 Joyent, Inc.
+ *
+ * `triton image update ...`
+ */
+
+var vasync = require('vasync');
+
+var common = require('../common');
+var errors = require('../errors');
+var UPDATE_IMAGE_FIELDS =
+    require('../cloudapi2').CloudApi.prototype.UPDATE_IMAGE_FIELDS;
+
+
+function do_update(subcmd, opts, args, cb) {
+    if (opts.help) {
+        this.do_help('help', {}, [subcmd], cb);
+        return;
+    }
+
+    var tritonapi = this.top.tritonapi;
+    var log = this.top.log;
+
+    if (args.length === 0) {
+        cb(new errors.UsageError('missing IMAGE argument'));
+        return;
+    }
+
+    var image = args.shift();
+
+    vasync.pipeline({arg: {cli: this.top}, funcs: [
+        common.cliSetupTritonApi,
+
+        function gatherDataArgs(ctx, next) {
+            try {
+                ctx.data = common.objFromKeyValueArgs(args, {
+                    disableDotted: true,
+                    typeHintFromKey: UPDATE_IMAGE_FIELDS
+                });
+            } catch (err) {
+                next(err);
+                return;
+            }
+
+            next();
+        },
+
+        function validateIt(ctx, next) {
+            try {
+                common.validateObject(ctx.data, UPDATE_IMAGE_FIELDS);
+            } catch (e) {
+                next(e);
+                return;
+            }
+
+            next();
+        },
+
+        function updateAway(ctx, next) {
+            log.trace({dryRun: opts.dry_run, fields: ctx.data},
+                'image update');
+
+            if (opts.dry_run) {
+                next();
+                return;
+            }
+
+            tritonapi.updateImage({
+                image: image,
+                fields: ctx.data
+            }, function (err, img) {
```

Comment:
How does one update an acl:
```
./bin/triton image update 506cdbc2 acl='[]'
triton image update: error (Usage): field "acl" must be of type "array", but got a value of type "object"
usage: triton image update IMAGE [FIELD=VALUE ...]
```

Reply:
My bad. The way `triton image share|unshare` works makes it way easier to handle `acl` using these commands. My idea was to exclude updating `acl` from the general update list, but I forgot to do so.